src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,38 +1,41 @@
-import React, { useState, useEffect } from "react";
+import React, { useEffect } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
-import { getTokenOwner } from "./../../../services/token";
-import { useEscrowContractUsers } from "../../../hooks";
-import { WrappedDocument } from "@govtechsg/open-attestation";
-
-interface TitleTransferPanelContentProps {
-  document: WrappedDocument;
-}
-
-export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
-  const [tokenOwnerAddress, setTokenOwnerAddress] = useState("");
-  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwnerAddress });
-
-  const fetchTokenOwner = async (document: WrappedDocument) => {
-    const address = await getTokenOwner({ document });
-    setTokenOwnerAddress(address);
-  };
+import { useDefaultProvider } from "../../../common/hooks/useDefaultProvider";
+import { useTitleEscrowContract } from "../../../common/hooks/useTitleEscrowContract";
+import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
+import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
 
+export const TitleTransferPanelContent = ({ titleEscrow }: { titleEscrow: TitleEscrow }) => {
+  const { call: getBeneficiary, value: beneficiary } = useContractFunctionHook(titleEscrow, "beneficiary");
+  const { call: getHolder, value: holder } = useContractFunctionHook(titleEscrow, "holder");
   useEffect(() => {
-    fetchTokenOwner(document);
-  }, [document]);
-
+    getBeneficiary();
+    getHolder();
+  }, [titleEscrow]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
why need to disable eslint ? does it trigger infinite render to add the functions ?

Reply:
needs to add `getBeneficiary` and `getHolder`. Both will trigger some infinite render and network calls.

![image](https://user-images.githubusercontent.com/7406870/77882471-95f76980-7293-11ea-9aba-64a6ac5e4dab.png)

Reply:
Also we cannot memoize the function from `useContractFunctionHook` because a calling a function twice with same params may result in different result. 

Ie. Issue a merkle root first time will go through, but the second time it should fail.

Reply:
hummmm this is problematic ...

Reply:
Fixed, by `useCallback` in ethers-contract-hook